lib/puppetserver/ca/action/list.rb

```diff
@@ -79,41 +88,62 @@ def run(input)
           requested, signed, revoked = separate_certs(all_certs)
           missing = certnames - all_certs.map { |cert| cert['name'] }
 
-          (all || certnames.any?) \
-            ? output_certs_by_state(requested, signed, revoked, missing)
-            : output_certs_by_state(requested)
+          (all || certnames.any?)\
+            ? output_certs_by_state(all, output_format, requested, signed, revoked, missing)
+            : output_certs_by_state(all, output_format, requested)
 
           return missing.any? \
             ? 1
             : 0
         end
 
-        def output_certs_by_state(requested, signed = [], revoked = [], missing = [])
+        def output_certs_by_state(all, output_format, requested, signed = [], revoked = [], missing = [])
+          grouped_cert = {}
           if revoked.empty? && signed.empty? && requested.empty? && missing.empty?
             @logger.inform "No certificates to list"
             return
           end
 
           unless requested.empty?
-            @logger.inform "Requested Certificates:"
-            output_certs(requested)
+            if output_format
+              all ? grouped_cert["requested"] = requested : @logger.inform(requested.to_json)
```

Comment:
This seems weird to me, or at least difficult to read.. imo it's confusing to have a single-line conditional where the two cases are doing totally different things, e.g. assigning a value vs. logging a message.